### DISCUSSION TITLE
How to get search parameters using API route (Solid Start)?

### DISCUSSION THREAD

**Discussion by benstahl** [solidjs/solid-router#258] [Q&A]
I'm trying to extract search parameters when an API route (GET) is called (I'm using Solid Start with TypeScript). However, I've found that calling useSearchParams() not only doesn't work, but it hangs the app in this setup. For example, I'm calling the API route with this URL and 2 search parameters:

`localhost:3001/chart9?test=2&another=blah`

and with this code in my (chart9).tsx file:

```
export async function GET() {
	const [searchParams, setSearchParams] = useSearchParams()
	console.log({ ...searchParams })

	return new Response('TEST')
}
```

The GET function is called correctly, but the app hangs on useSeachParams().

If I put it inside a component instead (same file, replacing the GET function), it works.

```
export default function MyComponent() {
	const [searchParams, setSearchParams] = useSearchParams()
	console.log({...searchParams})

	return (
		<main>
			<h1>TEST</h1>

		</main>
	)
}
```
and in the console, prints `{ test: '2', another: 'blah' }`, as expected.

Honestly, I don't care about useSearchParams() specifically working in the GET function, but how can I extract the search parameters from an API (ie. non-component) route? I can get the whole URL called by using `arguments['0'].request.url` inside GET, and then use another library to parse out the args, but that seems wasteful since Solid probably already knows what they are?

**Comment by crusty-code** [comment]
Hi,
I know i'm a little late but I had the same issue. I looked around on the internet but didn't find much. After some time looking around a realized that you can just wrap the url in a URL object like this:
`export function GET(e: APIEvent) {
    const url = new URL(e.request.url)

    console.log(url.search);
   
    return new Response("hello world")
}`
that gives you the searchParams, that you can feed into the `URLSearchParams` in javascript an you're done. Final code:

`export function GET(e: APIEvent) {
    const url = new URL(e.request.url)
    const searchParams = new URLSearchParams(url.search)
    
    console.log(searchParams.get("example"));

    return new Response("hello world")
}`
